src/components/EmailForm/index.tsx

```diff
@@ -0,0 +1,115 @@
+'use client';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useState } from 'react';
+import cx from 'classnames';
+
+import { Textarea } from '../Textarea';
+import { emailSchema } from './schema';
+
+type EmailBody = {
+  email_body: string;
+};
+
+type EmailFormProps = {
+  eventName?: string;
+};
+
+export const EmailForm: React.FC<EmailFormProps> = ({ eventName }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [emailData, setEmailData] = useState({
+    subject: '',
+    mail: '',
+    body: { email_body: '' },
+  });
+
+  const formProps = useForm<EmailBody>({
+    resolver: yupResolver(emailSchema),
+    defaultValues: {
+      email_body: '',
+    },
+  });
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = formProps;
+
+  const onSubmit = async (body: EmailBody) => {
+    const subject = `Registrace na sportovní akci ${eventName}`;
+    const mail = 'john.doe';
+
+    const encodedSubject = encodeURIComponent(subject);
+    const encodedBody = encodeURIComponent(body.email_body);
+
+    setEmailData({ subject, mail, body });
+
+    const mailLink = document.createElement('a');
+    mailLink.href = `mailto:${mail}?subject=${encodedSubject}&body=${encodedBody}`;
```

Comment:
This is definitely an interesting idea! But it leaks user email! :D Also, it opens terrible email clients (for me, it is Apple email, which I don't use) 

I would add this as an option to send email via some email icon (no prewriting of the message needed). <- On second thought, we don't know the owner's email. You can't make that public without consent

We still somehow need the "Ask the owner to reveal his email" or something like that. Old Sportujspolu (not my site) had the emails in plain text. I've copied many of them, in case we want to contact the ppl :D. But it is a horrible practice to have it that way

Reply:
and it might even be a GDPR issue.

Reply:
@GLObus303  well, if I will not show their emails, I need BE solution then, help! 

GDPR I can solve it with checkbox consent when user register? I also used fallbeck if users mail client does not work... so I  will revisit the solution if I have some BE help ready 🤓

Reply:
Even with consent, which everybody will blindly check, bots crawling will scrape the user's email. That wouldn't be nice from our side, right?

Reply:
Agreed. We are nice people.

Reply:
Noted for the next PR.